__tests__/pathfinding.test.js

```diff
@@ -0,0 +1,61 @@
+import * as testData from "../testData/pathfinding.data";
+import { POIInfo } from "../src/constants/poi.data";
+import { findPathOnFloor } from "../src/services/pathfinding.service";
+import * as floors from "../src/constants/floors.data";
+
+describe("Find the shortest path on a given floor", () => {
+  it("should return the shortest path between two given locations on H 8th floor", () => {
+    const shortest = findPathOnFloor(
+      1,
+      POIInfo.filter(({ displayName }) => displayName === "H851.03")[0]
+        .location,
+      POIInfo.filter(({ displayName }) => displayName === "H815")[0].location
```

Comment:
We will convert it to typescript if it's necessary. And indeed, if the floor data changes, the tests need to be changed. However, since the floor node data is hardcoded, and should not be changed, I don't think it will be a problem. An alternative will be mocking the node data in floors.data and have it as a fixture. Let me know which one you think is best.

Reply:
All the other tests are already js. Should we open another ticket to change all tests to ts?

Reply:
a ticket for that might be good. We should get on it really quickly though

Reply:
Ritchelle said (and i agree), that we should probably port tests to ts next sprint, and worry about mocking data probably after that port. I will be resolving this comment